enterprise/mqtt/src/main/java/io/crate/mqtt/protocol/MqttProcessor.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.mqtt.protocol;
+
+import io.crate.mqtt.netty.NettyUtils;
+import io.crate.mqtt.operations.MqttIngestService;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.handler.codec.mqtt.MqttConnectMessage;
+import io.netty.handler.codec.mqtt.MqttConnectReturnCode;
+import io.netty.handler.codec.mqtt.MqttPublishMessage;
+import io.netty.handler.codec.mqtt.MqttQoS;
+import io.netty.handler.codec.mqtt.MqttVersion;
+import io.netty.handler.timeout.IdleStateHandler;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.index.engine.VersionConflictEngineException;
+
+import java.util.Locale;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+
+
+public class MqttProcessor {
+
+    private static final Logger LOGGER = Loggers.getLogger(MqttProcessor.class);
+
+
+    private final MqttIngestService ingestService;
+
+    public MqttProcessor(MqttIngestService ingestService) {
+        this.ingestService = ingestService;
+    }
+
+    public void handleConnect(Channel channel, MqttConnectMessage msg) {
+        String clientId = msg.payload().clientIdentifier();
+        LOGGER.info("CONNECT for client <{}>", clientId);
+
+        if (!(msg.variableHeader().version() == MqttVersion.MQTT_3_1.protocolLevel()
+                || msg.variableHeader().version() == MqttVersion.MQTT_3_1_1.protocolLevel())) {
+            sendErrorResponse(channel, MqttConnectReturnCode.CONNECTION_REFUSED_UNACCEPTABLE_PROTOCOL_VERSION)
+                    .addListener(f -> LOGGER.warn("CONNECT sent UNACCEPTABLE_PROTOCOL_VERSION"));
+            return;
+        }
+
+        /*
+         If you don’t need a state to be hold by the broker,
```

Comment:
Did we write this comment or is this copied from somewhere? I don't understand this.

Reply:
It's a snippet from the mentioned http://www.hivemq.com/blog/mqtt-essentials-part-3-client-broker-connection-establishment which provides more context about clientid clean session